### DISCUSSION TITLE
e3nn for tensorflow?

### DISCUSSION THREAD

**Discussion by PatReis** [e3nn/e3nn#383] [Q&A] [answered]
Hallo, thanks for developing and publishing the package!

I saw that there is a jax version of e3nn in https://github.com/e3nn/e3nn-jax
Do you think it is possible to also have a tensorflow translation of e3nn for something like e3nn-tf or does this library involve too much python integration, and hence requires a fully dynamic computational graph, clashing with tensorflow?
If it were possible, are the developers willing to open a e3nn-tf repo?
I would be willing to contribute.

Thanks in advance for any replies or comments.
Cheers

**Comment by mariogeiger** [comment, marked as answer]
I unfortunately don't plan to port e3nn to tensorflow.
Why do you need to use tensorflow?

**Reply by PatReis** [reply]
I am collecting graph neural networks in tensorflow in https://github.com/aimat-lab/gcnn_keras . There is a little interest from industry and some implementation ideas within tensorflow. So I was wondering if e3nn could in principle be realised in tensorflow? I think your original TFN paper was implemented in tensorflow, right?
If you say it is possible I can give it a try to port to tensorflow. But if you do not want to have a tf-version to maintain (which is just more work) then this is okay and completely understandable.
I can see that the community in geometric deep learning has fully moved to pytorch or jax, at least in academics.

**Reply by PatReis** [reply]
Okay, maybe I was not clear. If I try to port to tensorflow, are you willing to support me? 
Meaning, I would transfer the repo to you if it is finished. You can still decide if you want to continue support or leave as is.

**Reply by mariogeiger** [reply]
Sure, worst case scenario it stays as is 😄 
Send me an email and I add you to the slack of e3nn (if it's not already done)

**Reply by PatReis** [reply]
Okay, thanks. I will send an e-mail.
